cfgov/unprocessed/apps/paying-for-college/js/disclosures/utils/promise-request.js

```diff
@@ -0,0 +1,30 @@
+/* Promise requests are better. */
+
+/**
+ * promiseRequest - A handy function for returning XHR Promises
+ * @param {string} method - The method, ex. POST or GET
+ * @param {string} url - The url to be requested
+ * @returns {object} Promise of the XHR request
+ */
+function promiseRequest(method, url) {
+  const xhr = new XMLHttpRequest();
+
+  return new Promise(function (resolve, reject) {
+    // Completed xhr
+    xhr.onreadystatechange = function () {
```

Comment:
You could consider using `fetch` instead of `xhr`, which may simplify the request code. See https://github.com/cfpb/consumerfinance.gov/pull/7653/files for example.

Reply:
Thanks for the tip, now that I'm closer to done with this, I'll check out `fetch`!

Reply:
PR https://github.com/cfpb/consumerfinance.gov/pull/8008